include/alpaka/kernel/TaskKernelCpuOmp2Blocks.hpp

```diff
@@ -99,7 +99,7 @@ namespace alpaka
                 // Get the size of the block shared dynamic memory.
                 auto const blockSharedMemDynSizeBytes(
                     meta::apply(
-                        [&](TArgs const & ... args)
+                        [&](typename std::decay<TArgs>::type const & ... args)
```

Comment:
May we be able to have a `using TArgsDecayed = typename std::decay<TArgs>::type`?

Reply:
`std::decay_t<>` is available with C++14. I am currently thinking of adding it in alpaka and switch with c++14 to the stl version.

Reply:
should I use `#if __cplusplus >= 201402L` to detect or is there a boost define available (I do not find any)

What I found on stackoverflow https://stackoverflow.com/questions/26089319/is-there-a-standard-definition-for-cplusplus-in-c14

Reply:
Having it as a typedef would still be less code in mot cases. Even with `decay_t` we would have to write it multiple time.
I am not sure if it is worth all the work when we switch to C++14 for the next alpaka release.